client/state/selectors/get-network-sites.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { isMainSiteToSecondarySiteConnection } from 'state/selectors';
+import { getSite } from 'state/sites/selectors';
+
+/**
+ * Returns the secondary sites of a given site
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  siteId      The ID of the site we're retrieving the secondary sites
+ * @return {?Array}            Array of secondary sites
```

Comment:
The result array will never be `null`, so let's remove the `null` notation from the JSdoc.